packages/@azure/servicebus/data-plane/test/sessionsTests.spec.ts

```diff
@@ -0,0 +1,494 @@
+    void
+  > {
+    await test_batching(unpartitionedQueueSessionClient, unpartitionedQueueSessionClient);
+  });
+
+  it("Unpartitioned Topics and Subscription with Sessions - Batch Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_batching(unpartitionedTopicSessionClient, unpartitionedSubscriptionSessionClient);
+  });
+
+  async function test_streaming(
+    senderClient: QueueClient | TopicClient,
+    sessionClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+
+    await senderClient.send(testMessagesWithSessions[0]);
+
+    let receiverClient = await sessionClient.acceptSession({sessionId:"non"+testSessionId});
+    let receivedMsgs: ServiceBusMessage[] = [];
+    receiverClient.receive((messageSession: MessageSession, msg: ServiceBusMessage) => {
+      receivedMsgs.push(msg);
+      return Promise.resolve();
+    }, unExpectedErrorHandler);
+    should.equal(receivedMsgs.length, 0);
+
+
+    receiverClient = await sessionClient.acceptSession({sessionId: testSessionId});
+    await testPeekMsgsLength(receiverClient, 1);
+    receivedMsgs = [];
+    receiverClient.receive((messageSession: MessageSession, msg: ServiceBusMessage) => {
+      receivedMsgs.push(msg);
+      should.equal(msg.body, testMessagesWithSessions[0].body);
+      should.equal(msg.messageId, testMessagesWithSessions[0].messageId);
+      return Promise.resolve();
+    }, unExpectedErrorHandler);
+    
+    for (let i = 0; i < 5; i++) {
+      await delay(1000);
+      if (receivedMsgs.length === testMessagesWithSessions.length) {
+        break;
+      }
+    }
+    should.equal(unexpectedError, undefined, unexpectedError && unexpectedError.message);
+
+    await testPeekMsgsLength(receiverClient, 0);
+  }
+
+
+  it("Partitioned Queues with Sessions - Streaming Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_streaming(partitionedQueueSessionClient, partitionedQueueSessionClient);
+  });
+
+  it("Partitioned Topics and Subscription with Sessions - Streaming Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_streaming(partitionedTopicSessionClient, partitionedSubscriptionSessionClient);
+  });
+
+  it("Unpartitioned Queues with Sessions - Streaming Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_streaming(unpartitionedQueueSessionClient, unpartitionedQueueSessionClient);
+  });
+
+  it("Unpartitioned Topics and Subscription with Sessions - Streaming Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_streaming(unpartitionedTopicSessionClient, unpartitionedSubscriptionSessionClient);
+  });
+});
+
+describe("Accept a session without passing sessionId and receive messages - Queue/Subscription has different messages belonging to different sessionId", function(): void {
```

Comment:
Rename to `Accept a session without passing sessionId and receive messages from randomly selected sessionId`